api/apps/articles/test/routes.test.coffee

```diff
@@ -108,11 +118,13 @@ describe 'routes', ->
 
   describe '#delete', ->
 
-    it 'delets an existing article', ->
+    it 'deletes an existing article', (done) ->
       @req.article = fixtures().articles
       routes.delete @req, @res
       @Article.destroy.args[0][1]()
-      @res.send.args[0][0].title.should.containEql 'Top Ten'
+      _.defer =>
+        @res.send.args[0][0].title.should.containEql 'Top Ten'
+        done()
```

Comment:
I think these `defer`'s are also not ideal but I couldn't think of a better way to do something like a `.then` or `await` on the router handler since it doesn't return a promise. When we modernize the route handler I can promisify/await on those in this test.

Reply:
So right now because of Coffeescript implicit returns, these functions actually do return promises since the last line is `present(article).then((results) => res.send(results))` and `present` is returning a promise.